.github/workflows/ci.yml

```diff
@@ -0,0 +1,409 @@
+      - run: docker images
+      - name: Upload Images
+        uses: actions/upload-artifact@v2
+        with:
+          name: docker-image-http-${{ matrix.nginx }}
+          path: ./tmp
+  build-prometheus-exporter-file:
+    name: Build prometheus-exporter-file
+    needs:
+      - lint-docker
+      - lint-shell
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - run: make build-prometheus-exporter-file
+      - run: cat ./tmp/image.tags | xargs -I % docker inspect --format='%={{.Id}}:{{index .Config.Env 7}}' %
+      - run: docker save "${DOCKER_IMAGE}" | gzip -9 > ./tmp/image-prometheus-exporter-file.tar
+      - run: docker images
+      - name: Upload Images
+        uses: actions/upload-artifact@v2
+        with:
+          name: docker-image-prometheus-exporter-file
+          path: ./tmp
+  build-php:
+    name: Build PHP ${{ matrix.php }} for ${{ matrix.type }} on Alpine ${{ matrix.alpine }}
+    needs:
+      - lint-docker
+      - lint-shell
+      - generate-alpine-strategy
+      - generate-php-strategy
+      - generate-php-type-strategy
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        alpine: ${{ fromJson(needs.generate-alpine-strategy.outputs.alpine) }}
+        php: ${{ fromJson(needs.generate-php-strategy.outputs.php) }}
+        type: ${{ fromJson(needs.generate-php-type-strategy.outputs.type) }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: ./build-php.sh ${{ matrix.type }} ${{ matrix.php }} ${{ matrix.alpine }}
+      - run: cat ./tmp/image.tags | xargs -I % docker inspect --format='%={{.Id}}:{{index .Config.Env 7}}' %
+      - run: docker save "${DOCKER_IMAGE}" | gzip -9 > ./tmp/image-${{ matrix.type }}-${{ matrix.php }}-${{ matrix.alpine }}.tar
+      - run: docker images
+      - name: Upload Images
+        uses: actions/upload-artifact@v2
+        with:
+          name: docker-image-${{ matrix.type }}-${{ matrix.php }}-${{ matrix.alpine }}
+          path: ./tmp
+  scan-vulnerability-php:
+    name: Scan PHP ${{ matrix.php }} for ${{ matrix.type }} on Alpine ${{ matrix.alpine }} for vulnerabilities
+    needs:
+      - build-php
+      - build-nginx
+      - build-prometheus-exporter-file
+      - generate-alpine-strategy
+      - generate-php-strategy
+      - generate-php-type-strategy
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        alpine: ${{ fromJson(needs.generate-alpine-strategy.outputs.alpine) }}
+        php: ${{ fromJson(needs.generate-php-strategy.outputs.php) }}
+        type: ${{ fromJson(needs.generate-php-type-strategy.outputs.type) }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install clair-scanner
+        run: |
+          sudo curl -L https://github.com/arminc/clair-scanner/releases/download/v8/clair-scanner_linux_amd64 -o /usr/local/bin/clair-scanner
+          sudo chmod +x /usr/local/bin/clair-scanner
```

Comment:
Have we considered using the [latest docker image](https://hub.docker.com/r/arminc/clair-local-scan) instead of having these installation steps in the various vulnerability jobs?

Reply:
I've mentioned that to CJ, but didn't take a look. Let me take a look, if it is easy, I'll switch to the docker image in this PR already.

Reply:
I've added some comments to the JIRA EPIC related to `php-docker-template`, the `clair` stuff needs some attention, not just here :)